picard/ui/metadatabox/tagdiff.py

```diff
@@ -342,3 +343,29 @@ def update_tag_names(self, changes_first=False, top_tags=None):
             self.tag_names = [
                 tag for tag in tag_names if
                 self.status[tag] != TagStatus.EMPTY]
+
+    def to_json(self):
+        result = {}
+        for tag in self.tag_names:
+            result[tag] = {}
+            if tag in self.old:
+                result[tag]['old'] = self.old[tag]
+            if tag in self.new:
+                result[tag]['new'] = self.new[tag]
+
+        return json.dumps(result)
+
+    def to_tsv(self):
+        COL_SEP = "\t"
+        LINE_SEP = "\n"
+        lines = []
+        for tag in self.tag_names:
+            columns = [
+                tag,
+                MULTI_VALUED_JOINER.join(self.old[tag]),
+                MULTI_VALUED_JOINER.join(self.new[tag])
+            ]
+            line = COL_SEP.join(columns).rstrip()
```

Comment:
Why `rstrip()` ? I don't think we need it.

Reply:
I've changed to using `csv.writer()` to handle all the secpaing. This leaves a trailing `\r\n` at the end of the string, which I don't like so I've kept one `rstrip()` right at the end